irpc.py

```diff
@@ -120,18 +178,60 @@ def add_provider_call(funcdef: FuncDef,
           for compound in l_compound:
               compound.block_items.insert(0, provider_call)
 
-def gen_adjacency_graph(l_provider, l_ent):
-    adjacency_graph = defaultdict(set)
+def gen_par_adjacency_graph(l_provider, l_ent):
+    par_adjacency_graph = defaultdict(set)
     for provdef in l_provider:
 
         entity = provider_name(provdef)
         entnames = l_ent - set([entity])
 
         # Insert the provider call
         for children_entity, _ in entity2Compound(provdef.body, entnames).items():
-            adjacency_graph[entity].add(children_entity)
+            par_adjacency_graph[entity].add(children_entity)
+
+    return par_adjacency_graph
+
+def gen_child_adjacency_graph(l_provider, l_ent):
+    child_adjacency_graph = defaultdict(set)
+
+    for provdef in l_provider:
+        entity = provider_name(provdef)
+        entnames = l_ent - set([entity])
+
+        for parent_entity, _ in entity2Compound(provdef.body, entnames).items():
+            child_adjacency_graph[parent_entity].add(entity)
 
-    return adjacency_graph
+    return child_adjacency_graph
+
+
+def find_touches(filename):
+    l_touch = set()
+    with open(filename, 'r') as input:
+        for line in input:
+            if "touch_" in line:
```

Comment:
Indeed, You can have false positive like
```
int my_variable_touch_a = 10
```
This is not a fun call, but will pass your test